### PR TITLE
Prevent collisions on let(:name) and let(:method_name)

### PR DESCRIPTION
**Pull request by benoittgt**
Fix: https://github.com/rspec/rspec-rails/issues/2451

Two commits. One to see the failures then one to see the fix. I did nothing specific, just implement the [great work](https://github.com/rspec/rspec-rails/issues/2451#issuecomment-771213822) of @grekko and the choice from @JonRowe.

### CODE REVIEW

File: lib/rspec/rails/fixture_support.rb [hunk 1]
```diff
@@ -9,11 +9,11 @@ module FixtureSupport
...
-        if ::Rails.version.to_f >= 6.1
-          # @private return the example name for TestFixtures
-          def name
-            @example
-          end
+        # @private prevent ActiveSupport::TestFixtures to start a DB transaction.
+        # Monkey patched to avoid collisions with 'let(:name)' in Rails 6.1 and after
+        # and let(:method_name) before Rails 6.1.
+        def run_in_transaction?
+          use_transactional_tests && !self.class.uses_transaction?(@example)
```
Comment (bcgraham) [inline lib/rspec/rails/fixture_support.rb:16]: Not “reviewing,” just curious - were you able to determine whether any RSpec code still sets `@example` to a meaningful value?

**Review by pirj — APPROVED** [review]
Awesome, thank you!

File: lib/rspec/rails/fixture_support.rb (hunk 1 above)
Comment (benoittgt) [reply lib/rspec/rails/fixture_support.rb:16]: No I didn't search this yet. But I will.

Comment (JonRowe) [reply lib/rspec/rails/fixture_support.rb:16]: For us this probably needs to be `example` or `self` as we don't assign `@example` within a test like mini test does afaik

**Comment by pirj** [comment]
Rebased. Green.
Apart from ruby: 2.2.10 with Rails `5-2-stable`, probably flaky. Passed on `pull_request` trigger, failed on `push`.

**Pull request merged**